tests/unit/packages/addons-service/provider-test.ts

```diff
@@ -0,0 +1,114 @@
+import { settled } from '@ember/test-helpers';
+import { setupMirage } from 'ember-cli-mirage/test-support';
+import { taskFor } from 'ember-concurrency-ts';
+import { setupTest } from 'ember-qunit';
+import { module, test } from 'qunit';
+
+import Provider from 'ember-osf-web/packages/addons-service/provider';
+import { CurrentUserStub } from 'ember-osf-web/tests/helpers/require-auth';
+
+module('Unit | Packages | addons-service | provider', function(hooks) {
+    setupTest(hooks);
+    setupMirage(hooks);
+
+    test('It works', async function(assert) {
+        const store = this.owner.lookup('service:store');
+        this.owner.register('service:current-user', CurrentUserStub);
+        const currentUser = this.owner.lookup('service:current-user');
+
+        const mirageUser = server.create('user', 'loggedIn');
+        const user = await store.findRecord('user', mirageUser.id);
+        currentUser.setProperties({ testUser: user, currentUserId: user.id });
+
+        const node = store.createRecord('node', {
+            id: 'wowza',
+            currentUserPermissions: [ 'read', 'write', 'admin' ],
+        });
+        // Make the addons service stuff
+        const externalStorageService = server.create('external-storage-service', {
+            id: 'bropdox',
+            name: 'Bropdox',
+        });
+        const internalResource = server.create('internal-resource', {
+            id: node.id,
+        });
+        const internalUser = server.create('internal-user', {
+            id: user.id,
+            configuredResources: [internalResource],
+            authorizedStorageAccounts: [],
+        });
+
+        server.create('configured-storage-addon', {
+            externalUserId: user.id,
+            externalUserDisplayName: user.fullName,
+            rootFolder: '/rooty-tooty/',
+            storageProvider: externalStorageService,
+            accountOwner: internalUser,
+            authorizedResource: internalResource,
+        });
+
+        const provider = new Provider(externalStorageService, currentUser, node);
+
+        assert.notOk(provider.internalUser, 'Provider internalUser is not set before initialize');
+        assert.notOk(provider.serviceNode, 'Provider serviceNode is not set before initialize');
+        assert.notOk(provider.configuredStorageAddon, 'Provider configuredStorageAddon is not set before initialize');
+
+        await settled();
```

Comment:
Proooobably won't have any timing issues related with doing it this way. I wonder if they ever made changes to ember if that would change, though.

Reply:
Yea, I'm not sure if these `notOk` asserts are really needed and they may eventually cause some weird breakage down the line if the test's async behavior changes. I think I will get rid of the `assert.notOk`s, as I don't think it's adding much value